core/model/modx/processors/resource/sort.class.php

```diff
@@ -39,6 +40,10 @@ public function process() {
         $this->getNodesFormatted($data, $this->getProperty('parent', 0));
 
         $this->fireBeforeSort();
+        if($this->getProperty('source_type')==="modResource"){
+            $resourceBeforeSort = $this->modx->getObject('modResource', array('id' => $this->getProperty('source_pk')));
+            $oldParentID = $resourceBeforeSort->get('parent');
```

Comment:
This could use a check if $resourceBeforeSort has a valid resource, otherwise this could cause a fatal error "method get on null" if for whatever reason the resource can't be loaded.